optimum/exporters/onnx/base.py

```diff
@@ -173,6 +173,14 @@ class OnnxConfig(ExportConfig, ABC):
                 "end_logits": {0: "batch_size", 1: "sequence_length"},
             }
         ),
+        "reinforcement-learning": OrderedDict(
+            {
+                "state_preds": {0: "batch_size", 1: "sequence_length"},
+                "action_preds": {0: "batch_size", 1: "sequence_length"},
+                "return_preds": {0: "batch_size", 1: "sequence_length"},
+                "last_hidden_state": {0: "batch_size", 1: "sequence_length"},
+            }
+        ),
```

Comment:
i think this is wrong as well, you'll need to investigate what are the actual dimensions and which ones are dynamic

Reply:
As mentioned in the output logs, output shapes are also matching based on model signatures.

	- Validating ONNX Model output "state_preds":
	- Validating ONNX Model output "state_preds":
	- Validating ONNX Model output "state_preds":
		-[✓] (2, 11, 11) matches (2, 11, 11)
		-[✓] (2, 11, 11) matches (2, 11, 11)
		-[✓] (2, 11, 11) matches (2, 11, 11)
		-[x] values not close enough, max diff: nan (atol: 1e-05)
		-[x] values not close enough, max diff: nan (atol: 1e-05)
		-[x] values not close enough, max diff: nan (atol: 1e-05)
	- Validating ONNX Model output "action_preds":
	- Validating ONNX Model output "action_preds":
	- Validating ONNX Model output "action_preds":
		-[✓] (2, 11, 3) matches (2, 11, 3)
		-[✓] (2, 11, 3) matches (2, 11, 3)
		-[✓] (2, 11, 3) matches (2, 11, 3)
		-[x] values not close enough, max diff: nan (atol: 1e-05)
		-[x] values not close enough, max diff: nan (atol: 1e-05)
		-[x] values not close enough, max diff: nan (atol: 1e-05)
	- Validating ONNX Model output "return_preds":
	- Validating ONNX Model output "return_preds":
	- Validating ONNX Model output "return_preds":
		-[✓] (2, 11, 1) matches (2, 11, 1)
		-[✓] (2, 11, 1) matches (2, 11, 1)
		-[✓] (2, 11, 1) matches (2, 11, 1)
		-[x] values not close enough, max diff: nan (atol: 1e-05)
		-[x] values not close enough, max diff: nan (atol: 1e-05)
		-[x] values not close enough, max diff: nan (atol: 1e-05)
	- Validating ONNX Model output "last_hidden_state":
	- Validating ONNX Model output "last_hidden_state":
	- Validating ONNX Model output "last_hidden_state":
		-[✓] (2, 33, 128) matches (2, 33, 128)
		-[✓] (2, 33, 128) matches (2, 33, 128)
		-[✓] (2, 33, 128) matches (2, 33, 128)
		-[x] values not close enough, max diff: nan (atol: 1e-05)
		-[x] values not close enough, max diff: nan (atol: 1e-05)
		-[x] values not close enough, max diff: nan (atol: 1e-05)

Reply:
yes they are right because onnx won't force the dynamic shapes on the model😅 but we need them to be correct for the model to handle dynamic shapes correctly. for example with the input `timesteps`, you specified a dynamic shape with `0: "batch_size", 1: "sequence_length"`, but that's wrong since you yourself defined the shape in the generator as:
```
        elif input_name == 'timesteps':
            shape = [self.normalized_config.config.state_dim, self.batch_size]
```
where batch size is the second axis and not the first (0).

Reply:
Thanks for the review comments. It's updated, please review.